astropy/modeling/astro_sherpa.py

```diff
@@ -0,0 +1,616 @@
+                      'moncar': MonCar, 'neldermead': NelderMead}
+
+
+class EstMethod(SherpaWrapper):
+
+    """
+    A wrapper for the error estimation methods of sherpa
+
+    Parameter:
+        value: String
+            the name of a sherpa statistics.
+    """
+
+    _sherpa_values = {'confidence': Confidence, 'covariance': Covariance,
+                      'projection': Projection}
+
+
+class SherpaFitter(Fitter):
+    __doc__ = """
+    Sherpa Fitter for astropy models. Yay :)
+
+    Parameters
+        ----------
+        optimizer : string
+            the name of a sherpa optimizer.
+            posible options include:
+                {opt}
+        statistic : string
+            the name of a sherpa statistic.
+            posible options include:
+                {stat}
+        estmethod : string
+            the name of a sherpa estmethod.
+            posible options include:
+                {est}
+    """.format(opt=", ".join(OptMethod._sherpa_values.keys()),
+               stat=", ".join(Stat._sherpa_values.keys()),
+               est=", ".join(EstMethod._sherpa_values.keys()))  # is this evil?
+
+    def __init__(self, optimizer="levmar", statistic="leastsq", estmethod="covariance"):
+        try:
+            optimizer = optimizer.value
+        except AttributeError:
+            optimizer = OptMethod(optimizer).value
+
+        try:
+            statistic = statistic.value
+        except AttributeError:
+            statistic = Stat(statistic).value
+
+        super(SherpaFitter, self).__init__(optimizer=optimizer, statistic=statistic)
+
+        try:
+            self._est_method = estmethod.value
+        except AttributeError:
+            self._est_method = EstMethod(estmethod).value
+
+        self.fit_info = {}
+        self._fitter = None  # a handle for sherpa fit function
+        self._fitmodel = None  # a handle for sherpa fit model
+        self._data = None  # a handle for sherpa dataset
+
+    def __call__(self, models, x, y, z=None, xerr=None, yerr=None, zerr=None, bkg=None, bkg_scale=1, **kwargs):
+        """
+        Fit the astropy model with a the sherpa fit routines.
+
+        Parameters
+        ----------
+        model : `~astropy.modeling.FittableModel`
+            model to fit to x, y, z
+        x : array
+            input coordinates
+        y : array
+            input coordinates
+        z : array (optional)
+            input coordinates
+        xerr : array (optional)
+            an array of errors in x
+        yerr : array (optional)
+            an array of errors in y
+        zerr : array (optional)
+            an array of errors in z
+        **kwargs:
+            keyword arguments will be passed on to sherpa fit routine
+
+        Returns
+        -------
+        model_copy : `~astropy.modeling.FittableModel` or a list of models.
+            a copy of the input model with parameters set by the fitter
+        """
+
+        tie_list = []
+        try:
```

Comment:
In these cases where the input could be a single object or a list of those objects, it can be simplest to just coerce to a list up front.  Then you don't need all the `if len(models) == 1` clauses below.  In some cases you need to remember that the input was a single object (`is_scalar` is a typical variable used for this) and de-list things at the end.

I guess in this case there is the question of whether inputting a single `model` or a `[model]` will do the same thing and have the same performance.  It should!  In this case you do something like:
```
if isinstance(models, astropy.modeling.FittableModel):
    models = [models]
```